app/src/main/java/org/jellyfin/mobile/fragment/ConnectFragment.kt

```diff
@@ -157,61 +162,54 @@ class ConnectFragment : Fragment() {
         connectionErrorText.isVisible = false
     }
 
-    private suspend fun checkServerUrlAndConnection(enteredUrl: String): HttpUrl? {
+    private suspend fun checkServerUrlAndConnection(enteredUrl: String): String? {
         Timber.i("checkServerUrlAndConnection $enteredUrl")
 
-        val normalizedUrl = enteredUrl.run {
-            if (lastOrNull() == '/') this
-            else "$this/"
-        }
-        val urls = jellyfin.discovery.addressCandidates(normalizedUrl)
-        Timber.i("Address candidates are $urls")
-
-        var httpUrl: HttpUrl? = null
-        var serverInfo: PublicSystemInfo? = null
-        loop@ for (url in urls) {
-            httpUrl = url.toHttpUrlOrNull()
+        val candidates = jellyfin.discovery.getAddressCandidates(enteredUrl)
+        Timber.i("Address candidates are $candidates")
 
-            if (httpUrl == null) {
-                showConnectionError(R.string.connection_error_invalid_format)
-                return null // Format is invalid, don't try any other variants
-            }
+        val recommendedServer = jellyfin.discovery.getRecommendedServer(candidates, false)
 
-            // Set API client address
-            apiClient.ChangeServerLocation(httpUrl.toString().trimEnd('/'))
+        // No server found that replied
+        if (recommendedServer == null) {
+            Timber.i("No recommended server found")
 
-            serverInfo = apiClient.getPublicSystemInfo()
-            if (serverInfo != null)
-                break@loop
+            //TODO add candidates to error
```

Comment:
Resolve TODO

Reply:
Could also simply ignore them, honestly. At least for now, until we migrate to Compose, then we could maybe draw a neat UI that shows a list of found but unavailable servers..

Reply:
We don't really show a useful error anymore now so I think we should at least do something